### PR TITLE
System binary

### PR DESCRIPTION
**Pull request by jtbairdsr**

### CODE REVIEW

**Comment by jtbairdsr** [comment]
I have pushed a change in response to your comments https://github.com/nodkz/mongodb-memory-server/issues/32#issuecomment-448137640.  I look forward to your feedback...

File: src/util/MongoBinary.js [hunk 1]
```diff
@@ -43,84 +122,60 @@ export default class MongoBinary {
...
-            : `MongoBinary: Download lock removed`
-        );
-      });
+    if (!binaryPath) {
+      binaryPath = await this.getDownloadPath(options);
     }
 
-    debug(`MongoBinary: Mongod binary path: ${this.cache[version]}`);
-    return this.cache[version];
+    if (version !== 'latest' && systemBinary) {
```
Comment (nodkz) [inline src/util/MongoBinary.js]: We should not show the message if versions are the same ;)

And I think that better to move this code inside `if (systemBinary) {` at line 149
right after `binaryPath = await this.getSystemPath(systemBinary);`

Comment (nodkz) [reply src/util/MongoBinary.js]: All rest parts are good! Code became much cleaner! Thanks!

Comment (nodkz) [reply src/util/MongoBinary.js]: Ouch! And don't forget to add new option to README.md

Comment (jtbairdsr) [reply src/util/MongoBinary.js]: Thanks, I appreciate the feedback and agree.  I have updated the code and the README.md.  let me know if there is anything else.

File: src/util/MongoBinary.js [hunk 2]
```diff
@@ -148,6 +148,21 @@ export default class MongoBinary {
 
     if (systemBinary) {
       binaryPath = await this.getSystemPath(systemBinary);
+      const binaryVersion = execSync('mongod --version')
```
Comment (nodkz) [inline src/util/MongoBinary.js]: Should be
```js
const binaryVersion = execSync(`${binaryPath} --version`)
```

And need to add check that `binaryPath` is not empty ;)

Comment (jtbairdsr) [reply src/util/MongoBinary.js]: oh yeah, good catch totally missed that.

Comment (nodkz) [reply src/util/MongoBinary.js]: I've just merged with the master (there was a conflict).
Pull changes before commit.

Comment (nodkz) [reply src/util/MongoBinary.js]: There were updated packages in master.
Maybe it helps to resolve #120

Comment (jtbairdsr) [reply src/util/MongoBinary.js]: 👌 I made that fix and pushed.

**Comment by nodkz** [comment]
Thanks a lot! 👍

**Comment by nodkz** [comment]
I'll fix
const binaryVersion = execSync(`${binaryPath} --version`)
myself

**Comment by jtbairdsr** [comment]
No problem! It was a pleasure to participate.

**Comment by jtbairdsr** [comment]
when will these changes be published to npm?

**Comment by jtbairdsr** [comment]
@nodkz thank you again for the awesome package!  It was a pleasure to contribute!

**Pull request merged**